### PR TITLE
Add support for fs.realpath.native in envs that support it

### PR DESCRIPTION
**Pull request by RyanZim**
Closes https://github.com/jprichardson/node-fs-extra/pull/650

### CODE REVIEW

**Comment by manidlou** [comment]
good! so we don't need to do anything for `fs.realpathSync.native`?

**Comment by RyanZim** [comment]
I think that should work without anything special, since we don't have promises in the mix, but it wouldn't hurt to add a test.

File: lib/fs/__tests__/realpath.test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+/* eslint-env mocha */
+
+// fs.realpath.native only available in Node v9.2+
+if (typeof fs.realpath.native === 'function') {
+  describe('realpath.native', () => {
+    it('works with callbacks', () => {
+      fse.realpath.native(__dirname, (err, path) => {
+        assert.ifError(err)
+        assert.equal(path, __dirname)
```
Comment (manidlou) [inline lib/fs/__tests__/realpath.test.js]: It is better to use `assert.strictEqual()` since `assert.equal()` is deprecated in the legacy mode.

Comment (manidlou) [inline lib/fs/__tests__/realpath.test.js]: It is better to use `assert.strictEqual()` since `assert.equal()` is deprecated in the legacy mode.

**Pull request merged**